assets/app/actions/siteActions.js

```diff
@@ -1,234 +1,132 @@
+      .then(updateRouterToSitesUri)
+      .catch(alertError);
   },
 
-  uploadFile(site, file, sha = false) {
-    const siteId = site.id;
-    const { name } = file;
-
-    convertFileToData(file).then((fileData) => {
-      const path = `assets/${name}`;
-      const message = `Uploads ${name} to project`;
-      let commit = {
-        content: fileData,
-        message: message
-      };
-
-      if (sha) commit = Object.assign({}, commit, { sha });
+  // todo rename to something like fetchTree
+  fetchFiles(site, path) {
+    return github.fetchRepositoryContent(site, path)
+      .then(dispatchSiteFilesReceivedAction.bind(null, site.id))
+      .catch(alertError);
+  },
 
-      return github.createCommit(site, path, commit);
-    }).then((commitObj) => {
-        alertActions.alertSuccess('File uploaded successfully');
-        this.fetchSiteAssets(site);
-    }).catch(error => alertActions.alertError(error.message));
+  fetchFileContent(site, path) {
+    return github.fetchRepositoryContent(site, path)
+      .then(dispatchSiteFileContentReceivedAction.bind(null, site.id));
   },
 
-  createPR(site, head, base) {
-    return github.createPullRequest(site, head, base).then((pr) => {
-      return github.mergePullRequest(site, pr);
-    }).then(() => {
-      return github.deleteBranch(site, head);
-    }).then(() => {
-      this.fetchBranches(site);
-    }).then(() => {
-      return alertActions.alertSuccess(`${head} merged successfully`);
-    }).catch(error => alertActions.httpError(error.message));
+  fetchSiteConfigs(site) {
+    return github.fetchRepositoryConfigs(site)
+      .then(dispatchSiteConfigsReceivedAction.bind(null, site.id))
+      .then(() => site);
   },
 
   createCommit(site, path, fileData, message = false, sha = false) {
-    const b64EncodedFileContents = encodeB64(fileData);
-    const siteId = site.id;
-    let commit = {
-      path,
-      message: (message) ? message : `Adds ${path} to project`,
-      content: b64EncodedFileContents,
-      branch: `${site.branch || site.defaultBranch}`,
-    };
-
-    if (sha) commit = Object.assign({}, commit, { sha });
+    const commit = addPathToSite(site, path, fileData, message, sha);
 
     return github.createCommit(site, path, commit).then((commitObj) => {
       alertActions.alertSuccess('File committed successfully');
-
-      store.dispatch({
-        type: siteActionTypes.SITE_FILE_CONTENT_RECEIVED,
-        siteId,
-        file: commitObj.content
-      });
-    }).catch(error => alertActions.httpError(error.message));
+      dispatchSiteFileContentReceivedAction(site.id, commitObj.content);
+    }).catch(alertError);
   },
 
   fetchSiteAssets(site) {
     const config = site['_config.yml'];
     const assetPath = (config && config.assetPath) || 'assets';
 
-    return github.fetchRepositoryContent(site, assetPath).then((assets) => {
+    const filterAssetsWithTypeOfFile = (assets) => {
       return assets.filter((asset) => {
```

Comment:
It could be. Would make the tests cleaner, and a cleaner test is generally indicative of clearer code.